### PR TITLE
fix: Handle 'edited' issue_comment webhooks for review detection [Midtown !1885]

### PR DESCRIPTION
**Pull request by btucker**
## Summary
- Reviewers often post a placeholder comment then edit it with the full review. The `edited` action was silently ignored in `handle_issue_comment`, so reviews were only detected by the polling fallback — meanwhile the nudge loop fired faster, causing repeated "please post your review" nudges.
- Now `edited` events are processed when the comment contains a review signature (`text_contains_review_signature`), matching the same path as `created` events. Non-review edits (typo fixes, etc.) are still ignored.

## Test plan
- [x] Added test for `edited` event with review signature → correctly detected as review
- [x] Added test for `edited` event without review signature → correctly ignored
- [x] All 2,106 existing tests pass
- [x] Clippy clean

🌃 Co-built with [Midtown](https://github.com/btucker/midtown)

### CODE REVIEW

File: src/webhook.rs [hunk 1]
```diff
@@ -870,7 +870,14 @@ fn handle_pull_request_review(body: &[u8]) -> Result<Option<WebhookEvent>, serde
...
 
-    if event.action != "created" {
+    // Process 'created' events always. Process 'edited' events only when
+    // the comment contains a review signature — reviewers often post a
+    // placeholder then edit it with the full review.
+    let is_edited = event.action == "edited";
+    if event.action != "created" && !is_edited {
+        return Ok(None);
+    }
+    if is_edited && !is_review_comment(&event.comment.body) {
```
[the remark this answers is not in the document]
Comment (btucker) [reply src/webhook.rs]: ✅ Addressed in 942dfc8. Now checks `changes.body.from` to distinguish non-review→review transitions from edits to already-posted reviews. Added tests for both cases plus the edge case where `changes` is absent.

**Comment by btucker** [comment]
✅ Addressed in a22df17 — extracted all ~57 inline webhook tests to a separate `src/webhook_tests.rs` file using the `#[path = "webhook_tests.rs"] #[cfg(test)] mod tests;` pattern per CLAUDE.md convention.

🌃 Co-built with [Midtown](https://github.com/btucker/midtown)
